lnwallet/btcwallet/btcwallet.go

```diff
@@ -1104,6 +1105,31 @@ func extractBalanceDelta(
 	return balanceDelta, nil
 }
 
+// getPreviousOutpoints is a helper function which gets the previous
+// outpoints of a transaction.
+func getPreviousOutpoints(wireTx *wire.MsgTx, myInputs []wallet.TransactionSummaryInput) []lnwallet.PreviousOutPoint {
+	// isOurOutput is a map containing the output indices
+	// controlled by the wallet.
+	// Note: We make use of the information in `myInputs` provided
+	// by the `wallet.TransactionSummary` structure that holds
+	// information only if the input/previous_output is controlled by the wallet.
+	isOurOutput := make(map[uint32]bool, len(myInputs))
+	for _, myInput := range myInputs {
+		txIn := wireTx.TxIn[myInput.Index]
+		isOurOutput[txIn.PreviousOutPoint.Index] = true
```

Comment:
I don't think we need to take the detour of using `txIn.PreviousOutPoint.Index` here. Can't we just directly set `isOurOutput[myInput.Index] = true` and then below use `IsOurOutput: isOurOutput[idx]`?